projects/element-translate-ng/ngx-translate/si-translate-ngxt.service.ts

```diff
@@ -2,34 +2,40 @@
  * Copyright (c) Siemens 2016 - 2025
  * SPDX-License-Identifier: MIT
  */
-import { Injectable } from '@angular/core';
 import {
-  MissingTranslationHandler,
   MissingTranslationHandlerParams,
-  TranslateService
+  TranslateParser,
+  TranslateService,
+  Translation
 } from '@ngx-translate/core';
 import { SiTranslateService, TranslationResult } from '@siemens/element-translate-ng/translate';
 import { merge, Observable, of } from 'rxjs';
 import { map } from 'rxjs/operators';
 
+interface MissingTranslationHandler {
+  handle: (params: MissingTranslationHandlerParams) => Translation | Observable<Translation>;
+}
+
 /**
  * {@link SiTranslateService} wrapper around ngx-translate
  *
  * @internal
  */
-@Injectable()
 export class SiTranslateNgxTService extends SiTranslateService {
   private ngxTranslateService: TranslateService;
+  private parser: TranslateParser;
   private defaultTranslations: Record<string, string> = {};
   private originalMissingHandler: MissingTranslationHandler;
 
   // eslint-disable-next-line @angular-eslint/prefer-inject
-  constructor(ngxTranslateService: TranslateService) {
+  constructor(ngxTranslateService: TranslateService, parser: TranslateParser) {
     super();
     this.ngxTranslateService = ngxTranslateService;
-    this.originalMissingHandler = ngxTranslateService.missingTranslationHandler;
-    ngxTranslateService.missingTranslationHandler = {
-      handle: params => this.handleMissingTranslation(params)
+    this.parser = parser;
+    this.originalMissingHandler = (ngxTranslateService as any)
+      .missingTranslationHandler as MissingTranslationHandler;
+    (ngxTranslateService as any).missingTranslationHandler = {
+      handle: (params: MissingTranslationHandlerParams) => this.handleMissingTranslation(params)
```

Comment:
Instead of using the private field, you can just inject the current MissingTranslationHandler